libs/utils/executor.py

```diff
@@ -477,21 +477,21 @@ def _wload_init(self, tc, wl_idx):
         wload = self._wload_conf(wl_idx, wlspec)
 
         # Keep track of platform configuration
-        self.te.test_dir = '{}/{}:{}:{}'\
+        test_dir = '{}/{}:{}:{}'\
             .format(self.te.res_dir, wload.wtype, tc_idx, wl_idx)
-        os.system('mkdir -p ' + self.te.test_dir)
-        self.te.platform_dump(self.te.test_dir)
+        os.system('mkdir -p ' + test_dir)
```

Comment:
Should we better use Python's APIs?
```python
try:
    os.stat(test_dir)
except:
    os.mkdir(test_dir)
```
I guess this should take care of potential spaces in the path.